dask/dataframe/io.py

```diff
@@ -665,8 +673,10 @@ def from_delayed(dfs, metadata=None, divisions=None, columns=None,
 
     if divisions is None:
         divisions = [None] * (len(dfs) + 1)
+    if metadata is None:
+        metadata = dfs[0].compute()
```

Comment:
Seems reasonable to me, but is there a potentially smaller object that could be kept in memory? Something like `dfs[0].compute().iloc[:0]`? Assuming that this result doesn't get re-used later so you do actually have to compute it twice.